addons/mass_mailing_sms/tests/test_mailing_unsubscribe.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from datetime import datetime
+
+from odoo.tests.common import users
+from odoo.addons.mass_mailing_sms.tests.common import MassSMSCommon
+import werkzeug.urls
+import random
+from odoo.tests import HttpCase
+
+class TestMailingListSms(HttpCase,MassSMSCommon):
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestMailingListSms, cls).setUpClass()
+        cls._create_mailing_list()
+
+    @users('user_marketing')
+    def test_mailing_list_action_send_sms(self):
+
+        partner = self.env['res.partner'].create({
+            'name': 'Test Partner',
+            'phone': '+91 1234657890',
+        })
+
+        mailing = self.env['mailing.mailing'].create({
+            'name': 'TestMailing',
+            'subject': 'Test',
+            'mailing_type': 'sms',
+            'body_plaintext': 'Coucou hibou',
+            'mailing_model_id': self.env['ir.model']._get('res.partner').id,
+            'contact_list_ids': [],
+            'sms_allow_unsubscribe': True,
+        })
+        mailing.action_send_sms()
+
+        trace = mailing.mailing_trace_ids.filtered(lambda t: t.res_id == partner.id)
+
+        self.assertTrue(mailing.id, 'Mailing ID not set')
+
+        unsubscribe_url = werkzeug.urls.url_join(
+            mailing.get_base_url(),
+            '/sms/%s/%s' % (mailing.id, trace.sms_code)
+        )
+        parsed_url = werkzeug.urls.url_parse(unsubscribe_url)
```

Comment:
Unused. And therefore `unsubscribe_url` as it is too. 
We could either check that it is in the sms body, but we could also add the `/unsubscribe` to it and use it with the opener?